common/djangoapps/terrain/stubs/edxnotes.py

```diff
@@ -191,6 +200,41 @@ def _delete(self, note_id):
         else:
             self.respond(404, "404 Not Found")
 
+    def _get_paginated_response(self, notes, page_num, page_size):
+        """
+        Returns a paginated response of notes.
+        """
+        start = (page_num - 1) * page_size
+        end = start + page_size
+        total_notes = len(notes)
+        url_path = "http://{server_address}:{port}{path}".format(
+            server_address=self.client_address[0],
+            port=self.client_address[1],
+            path=self.path_only
+        )
+
+        next_url = 'null' if end >= total_notes else _get_next_prev_url(
+            url_path, deepcopy(self.get_params), page_num + 1, page_size
+        )
+        prev_url = 'null' if page_num == 1 else _get_next_prev_url(
+            url_path, deepcopy(self.get_params), page_num - 1, page_size)
```

Comment:
It would be good if you can move above `deepcopy` calls into `_get_next_prev_url`